QuasistaticBrownianThermalNoise/QuasistaticBrownianThermalNoise.cpp

```diff
@@ -642,11 +756,41 @@ CylTransFunc::CylTransFunc(double coatThick, double halfCylThick):
 	      Y_Iso_AlGaAs[i][j][k][l] += lame_mu_AlGaAs;
 	    }
 	    
-	  }
-	}
+              }
+          }
       }
     }  
 
+      // Here we use mu = lambda_mu_AlGaAs. To solve for the bulk modulus, we use
+      // K = lambda + 2/3 mu
+      // where K is the bulk modulus, lambda is the first lame coefficient, and mu is the shear modulus/ second lame coeffiecient.
+      
+      // Y[i][j][k][l]
+      
+      for(int i=0;i<dim;++i) {
+          for(int j=0;j<dim;++j) {
+              for(int k=0;k<dim;++k) {
+                  for(int l=0;l<dim;++l) {
+                      
+                      //Y_{ijkl} = lambda dij dkl + mu dik djl + mu dil djk
+                      if(i==j && k==l)
+                      {
+                          Y_Iso_AlGaAs_Phi[i][j][k][l] += (((lame_lambda_AlGaAs + (lame_mu_AlGaAs * (2.0/3.0)))  * lossPhi_Iso_AlGaAs_bulk) - ((2.0/3.0) * lame_mu_AlGaAs * lossPhi_Iso_AlGaAs_shear));
+                      }
+                      
+                      if(i==k && j==l) {
+                          Y_Iso_AlGaAs_Phi[i][j][k][l] += (lame_mu_AlGaAs * lossPhi_Iso_AlGaAs_shear);
+                      }
+                      
+                      if(i==l && j==k) {
+                          Y_Iso_AlGaAs_Phi[i][j][k][l] += (lame_mu_AlGaAs  * lossPhi_Iso_AlGaAs_shear);
+                      }
+                      
+                  }
+              }
+          }
+      }
+      
```

Comment:
Same as above: don't do this out separately for each material. Make a little helper function todo it for any lame_lambda, lame_mu, loss_phi_bulk, and loss_phi_shear. Then call the function for each material.